Automattic/jetpack
# Social: Add toggle to enable or disable Publicize

This PR adds a simple toggle to enable or disable the Publicize module from Jetpack Social's admin page.

#### Changes proposed in this Pull Request:

* Add a toggle to enable or disable Publizice.
* Add a store to the plugin, and refactor the connections list to use this store.
* Add endpoints to update Jetpack (module) settings.
* Add functions to the `api` package to fetch and update Jetpack Social settings.


https://user-images.githubusercontent.com/1713699/166208577-e795526d-bf61-4f8c-a0f8-29af1576262c.mp4


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






(see video for demonstration)

* Go to the Jetpack Social admin page.
* Confirm the current status of the module by running `jetpack docker wp option get jetpack_active_modules`.
* Use the toggle to enable or disable Publicize, and ensure the change went through by running the command again.
* Refresh the page with either the module enabled or disabled, and ensure the toggle is properly enabled or disabled.
* If you have access to a sandbox, check that the `active_modules` setting is synced correctly.
